sdk/core/core-http/lib/policies/proxyPolicy.ts

```diff
@@ -26,21 +31,32 @@ function loadEnvironmentProxyValue(): string | undefined {
   return undefined;
 }
 
-export function getDefaultProxySettings(proxyUrl?: string): ProxySettings | undefined {
+export function getDefaultProxySettings(
+  proxyUrl?: string,
+  username?: string,
+  password?: string
+): ProxySettings | undefined {
   if (!proxyUrl) {
     proxyUrl = loadEnvironmentProxyValue();
```

Comment:
How does this work with environment proxies?

Reply:
Oh. I'm not aware of the internal details.

Reply:
Implementation of `loadEnvironmentProxyValue()`
https://github.com/Azure/azure-sdk-for-js/blob/87a13978ccbc404aef035b92bf2a44d960fbc2d3/sdk/core/core-http/lib/policies/proxyPolicy.ts#L16-L32

It is not obvious on how(or why) someone would set username/password as environment variables to load them from the environment with `loadEnvironmentProxyValue`.

We can go for some syntax like `http://username:password@ipaddress:port` for the proxyUrl variable and parse it properly.

Or we can avoid using `getDefaultProxySettings` like mentioned below -   https://github.com/Azure/azure-sdk-for-js/pull/5009#discussion_r321611177.

Thoughts?

Reply:
Looking at [the .NET tests](https://github.com/Azure/azure-sdk-for-net/blob/a866db7ee9b64eef132f8e67acb542ff36e813ab/sdk/core/Azure.Core/tests/HttpEnvironmentProxyTest.cs#L134) they do support urls with credentials.

Reply:
[the relevant .NET commit](https://github.com/Azure/azure-sdk-for-net/commit/a866db7ee9b64eef132f8e67acb542ff36e813ab)

Reply:
Supporting parsing `http://username:password@ipaddress:port` into a ProxySettings object seems like the right approach here.

> It is not obvious on how(or why) someone would set username/password as environment variables to load them from the environment with loadEnvironmentProxyValue.

username/password are simply part of the proxy URL. One scenario I've seen many times is the proxy settings aren't controlled by the app developer but instead by the ops folks who control the production/staging environment. Devs just send requests like normal, ops folks ensure the environment variable is set properly, and everything "just works (tm)".

Reply:
I agree with @bterlson, but would keep parsing username/password from the url out of this PR and out of the current preview.